### PR TITLE
Tide: Some small codesimplifications that do not change behavior

### PR DESCRIPTION
**Pull request by alvaroaleman**
/cc @chaodaiG

### CODE REVIEW

File: prow/tide/tide.go [hunk 1]
```diff
@@ -1610,7 +1597,7 @@ func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit,
 				continue
 			}
 
-			record(int(pr.Number), ps)
+			presubmits[int(pr.Number)] = append(presubmits[int(pr.Number)], ps)
```
Comment (alvaroaleman) [inline prow/tide/tide.go:1600]: It is perfectly valid to append to a nil slice, the previous code was needlessly complex and annoying to read, as you had to go back up to read the inline func

File: prow/tide/tide.go [hunk 2]
```diff
@@ -919,9 +919,7 @@ func (c *Controller) accumulateBatch(sp subpool) (successBatch []PullRequest, pe
 		context := pj.Spec.Context
 		jobState := toSimpleState(pj.Status.State)
 		// Store the best result for this ref+context.
-		if s, ok := states[ref].jobStates[context]; !ok || s == failureState || jobState == successState {
```
Comment (alvaroaleman) [inline prow/tide/tide.go:922]: The implementation of `getBetterSimpleState` basically matches what is here. Funnily, the two inline checks that did this previously did it in a different ways (but both correct).

File: prow/tide/tide.go [hunk 3]
```diff
@@ -2263,3 +2250,13 @@ func mapKeyWithHighestvalue(m map[string]int) string {
 
 	return result
 }
+
+func getBetterSimpleState(a, b simpleState) simpleState {
```
Comment (chaodaiG) [inline prow/tide/tide.go:2254]: nit: would be nice to mention here that only "success, pending, failure" states are supported statues

File: prow/tide/tide.go [hunk 4]
```diff
@@ -2263,3 +2250,13 @@ func mapKeyWithHighestvalue(m map[string]int) string {
...
+	if a == "" || a == failureState || b == successState {
+		// b can't be worse than no state or failure and a can't be beter than success
+		return b
+	}
+
+	// a must be pending and b can not be success, so b can't be better than a
```
Comment (chaodaiG) [inline prow/tide/tide.go:2260]: I like this kind of math problem assertion :)

**Review by chaodaiG — APPROVED** [review]
/hold
Looks good to me, left a nit, feel free unhold if you don't want to address my nit

File: prow/tide/tide.go (hunk 3 above)
Comment (alvaroaleman) [reply prow/tide/tide.go:2256]: Updated, wdyt?

**Review by chaodaiG — APPROVED** [review]
/unhold

**Pull request merged**